streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -100,7 +100,7 @@ class Subscription(
   def sendMessage(msg: Message, partition: Int): Int = {
 
     // only send message whose timestamp matches the lifeTime
-    if (life.contains(msg.timestamp)) {
+    if (partition >= 0 && life.contains(msg.timestamp)) {
```

Comment:
gearpump broadcasts messages to all subscribers which will cause error at storm's side. I'm trying to solve this by returning partition -1 for those unintended subscribers and filter it out when `sendMessage`